cpu/kinetis_common/include/periph_cpu.h

```diff
@@ -41,19 +41,9 @@ typedef uint16_t gpio_t;
 #define GPIO_UNDEF          (0xffff)
 
 /**
- * @brief   Definition of pin and port positions in the gpio_t type
- * @{
- */
-#define GPIO_PORT_SHIFT     (8)
-#define GPIO_PORT_MASK      (0xff << (GPIO_PORT_SHIFT))
-#define GPIO_PIN_SHIFT      (0)
-#define GPIO_PIN_MASK       (0xff << (GPIO_PIN_SHIFT))
-/** @} */
-
-/**
  * @brief   Define a CPU specific GPIO pin generator macro
  */
-#define GPIO_PIN(port, pin)          ((port << GPIO_PORT_SHIFT) | pin)
+#define GPIO_PIN(x, y)      (((x + 1) << 12) | (x << 6) | y)
```

Comment:
I understand now, but I think we need at least an explanation inside a code comment.
However, do we really need to have both x and x+1 inside the pin value? Would it not make more sense to just do `+ 1` inside the port function below?